File: EventshopJava/src/main/webapp/proc/src/emage.cc
#include "op.h"


Emage::Emage()
{
	theme = "";

	start_time = end_time = 0;
	lat_unit = long_unit = 0;
	sw_lat = sw_long = ne_lat = ne_long = 0;
	num_rows = num_cols = 0;
}


// Shallow Initialization
Emage::Emage(const Emage& emage) : array(emage.array)
{
	theme = emage.theme;

	start_time = emage.start_time;
	end_time = emage.end_time;

	lat_unit = emage.lat_unit;
	long_unit = emage.long_unit;
	sw_lat = emage.sw_lat;
	sw_long = emage.sw_long;
	ne_lat = emage.ne_lat;
	ne_long = emage.ne_long;

	num_rows = emage.num_rows;
	num_cols = emage.num_cols;
}


// Deep Initialization
Emage::Emage(const Emage& emage, const cv::Mat& m)
{
	theme = emage.theme;

	start_time = emage.start_time;
	end_time = emage.end_time;

	lat_unit = emage.lat_unit;
	long_unit = emage.long_unit;
	sw_lat = emage.sw_lat;
	sw_long = emage.sw_long;
	ne_lat = emage.ne_lat;
	ne_long = emage.ne_long;

	num_rows = emage.num_rows;
	num_cols = emage.num_cols;

	array = m.clone();
}


int64 Emage::getEndTime() const
{
    return end_time;
}

double Emage::getLatUnit() const
{
    return lat_unit;
}

double Emage::getLongUnit() const
{
    return long_unit;
}

double Emage::getNeLat() const
{
    return ne_lat;
}

double Emage::getNeLong() const
{
    return ne_long;
}

int32 Emage::getNumCols() const
{
    return num_cols;
}

int32 Emage::getNumRows() const
{
    return num_rows;
}

int64 Emage::getStartTime() const
{
    return start_time;
}

double Emage::getSwLat() const
{
    return sw_lat;
}

double Emage::getSwLong() const
{
    return sw_long;
}

string Emage::getTheme() const
{
    return theme;
}

void Emage::setEndTime(const int64 end_time)
{
    this->end_time = end_time;
}

void Emage::setLatUnit(const double lat_unit)
{
    this->lat_unit = lat_unit;
}

void Emage::setLongUnit(const double long_unit)
{
    this->long_unit = long_unit;
}

void Emage::setNeLat(const double ne_lat)
{
    this->ne_lat = ne_lat;
}

void Emage::setNeLong(const double ne_long)
{
    this->ne_long = ne_long;
}

void Emage::setNumCols(const int32 num_cols)
{
    this->num_cols = num_cols;
}

void Emage::setNumRows(const int32 num_rows)
{
    this->num_rows = num_rows;
}

void Emage::setStartTime(const int64 start_time)
{
    this->start_time = start_time;
}

void Emage::setSwLat(const double sw_lat)
{
    this->sw_lat = sw_lat;
}

void Emage::setSwLong(const double sw_long)
{
    this->sw_long = sw_long;
}

void Emage::setTheme(const string& theme)
{
    this->theme = string(theme);
}

File: EventshopJava/src/main/java/com/eventshop/eventshoplinux/model/Emage.java

package com.eventshop.eventshoplinux.model;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;

import java.util.List;

/**
 * Created by aravindh on 5/12/15.
 */
@JsonIgnoreProperties
public class Emage {
    public String theme;
    public long startTime;
    public long endTime;
    public String startTimeStr;
    public String endTimeStr;
    public double latUnit; // Resolution of latitude
    public double longUnit; // Resolution of longitude
    public double swLat;
    public double swLong;
    public double neLat;
    public double neLong;
    public int row = 0;
    public int col = 0;
    public double min = 0;
    public double max = 0;
    public double[] image;
    public String value;
    public List<String> colors;
    String mapEnabled;

    public Emage() {

    }

    public Emage(String theme, long startTime, long endTime, String startTimeStr, String endTimeStr,
                 double latUnit, double longUnit, double swLat, double swLong, double neLat, double neLong,
                 int row, int col, double min, double max, double[] image, String value, List<String> colors, String mapEnabled) {
        this.theme = theme;
        this.startTime = startTime;
        this.endTime = endTime;
        this.startTimeStr = startTimeStr;
        this.endTimeStr = endTimeStr;
        this.latUnit = latUnit;
        this.longUnit = longUnit;
        this.swLat = swLat;
        this.swLong = swLong;
        this.neLat = neLat;
        this.neLong = neLong;
        this.row = row;
        this.col = col;
        this.min = min;
        this.max = max;
        this.image = image;
        this.value = value;
        this.colors = colors;
        this.mapEnabled = mapEnabled;
    }

    public String getTheme() {
        return theme;
    }

    public void setTheme(String theme) {
        this.theme = theme;
    }

    public long getStartTime() {
        return startTime;
    }

    public void setStartTime(long startTime) {
        this.startTime = startTime;
    }

    public long getEndTime() {
        return endTime;
    }

    public void setEndTime(long endTime) {
        this.endTime = endTime;
    }

    public double getLatUnit() {
        return latUnit;
    }

    public void setLatUnit(double latUnit) {
        this.latUnit = latUnit;
    }

    public double getLongUnit() {
        return longUnit;
    }

    public void setLongUnit(double longUnit) {
        this.longUnit = longUnit;
    }

    public double getSwLat() {
        return swLat;
    }

    public void setSwLat(double swLat) {
        this.swLat = swLat;
    }

    public double getSwLong() {
        return swLong;
    }

    public void setSwLong(double swLong) {
        this.swLong = swLong;
    }

    public double getNeLat() {
        return neLat;
    }

    public void setNeLat(double neLat) {
        this.neLat = neLat;
    }

    public double getNeLong() {
        return neLong;
    }

    public void setNeLong(double neLong) {
        this.neLong = neLong;
    }

    public int getRow() {
        return row;
    }

    public void setRow(int row) {
        this.row = row;
    }

    public int getCol() {
        return col;
    }

    public void setCol(int col) {
        this.col = col;
    }

    public double[] getImage() {
        return image;
    }

    public void setImage(double[] image) {
        this.image = image;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public List<String> getColors() {
        return colors;
    }

    public void setColors(List<String> colors) {
        this.colors = colors;
    }

    public String getMapEnabled() {
        return mapEnabled;
    }

    public void setMapEnabled(String mapEnabled) {
        this.mapEnabled = mapEnabled;
    }

    public double getMin() {
        return min;
    }

    public void setMin(double min) {
        this.min = min;
    }

    public double getMax() {
        return max;
    }

    public void setMax(double max) {
        this.max = max;
    }

    public String getStartTimeStr() {
        return startTimeStr;
    }

    public void setStartTimeStr(String startTimeStr) {
        this.startTimeStr = startTimeStr;
    }

    public String getEndTimeStr() {
        return endTimeStr;
    }

    public void setEndTimeStr(String endTimeStr) {
        this.endTimeStr = endTimeStr;
    }


}
